### ISSUE TITLE
Advanced SSA Fonts/Typesetting are not displaying correctly

### ISSUE THREAD

**Issue by ZhunCn** [jasongdove/ErsatzTV#1044]
Anime SSA subtitles often have custom fonts/typesetting that alter the video (transformation, masking, etc.)
PR #742 had extracting subtitles from the original video file integrated, but commented out [here](https://github.com/jasongdove/ErsatzTV/blob/9a30d7c7da5acb2561cde0bcd5d45f5f02aeeffb/ErsatzTV.Application/Subtitles/Commands/ExtractEmbeddedSubtitlesHandler.cs#L153).

When I built it with the line uncommented and `episodeId` changed to `mediaItemId`, the subtitles does work properly with the correct fonts now being burned into the video stream.

```
await ExtractFonts(dbContext, mediaItemId, ffmpegPath, cancellationToken);
```

Is there a reason why it was commented out? It looks like it was working completely fine.

**Comment by jasongdove** [comment]
My initial guess is that I was unhappy with dumping all attachments in a fonts folder (not just fonts). Perhaps I can run a cleanup process after extraction.

Will revisit this soon, thanks for the detailed direction.

**Comment by jasongdove** [comment]
@ZhunCn I think I found a way to filter (ignore non-font) attachments using the mime type here:

https://github.com/jasongdove/ErsatzTV/blob/1d90070cc7d8f9bef73806e4555b4ea5d363ab08/ErsatzTV.Application/Subtitles/Commands/ExtractEmbeddedSubtitlesHandler.cs#L354-L355

Will that work for your purposes? I don't have a lot of test content with attached fonts, but it looks ok with the few I do have. You can `ffprobe` to check for mime types, something like

```shell
ffprobe -v error -show_streams filename.mkv | grep mimetype | sort | uniq
```

**Comment by ZhunCn** [comment]
For various different movies and tv shows, I've only had this pop up.
* `TAG:mimetype=application/vnd.ms-opentype`
* `TAG:mimetype=application/x-truetype-font`
* `TAG:mimetype=font/ttf`
* `TAG:mimetype=font/otf`
I think it is covered by your PR. So LGTM. Are you going to look at cleaning up the files, or just leave it in the fonts directory?

**Comment by jasongdove** [comment]
I tweaked the logic so it only extracts font attachments, so there shouldn't be anything to cleanup. Or do you mean cleanup the font files at some point after the item has played?

**Comment by ZhunCn** [comment]
I was thinking the latter because of your initial comment, but if it is not an issue, then I have no problems here.

**Comment by jasongdove** [comment]
Ah, perhaps I didn't explain clearly.

The commented-out code extracted all attachments from files that contained text subtitles, even things like pictures which wouldn't make sense in a fonts folder. I was thinking I could extract everything and then clean up anything that wasn't a font.

When I took a closer look today, I saw that I could extract each font individually so there is no longer any worry about pictures etc making it into the fonts cache.
